Assets/MRTK/Services/InputSystem/FocusProvider.cs

```diff
@@ -1248,88 +1246,23 @@ private static void QueryScene(IMixedRealityPointer pointer, IMixedRealityRaycas
                     switch (pointer.SceneQueryType)
                     {
                         case SceneQueryType.SimpleRaycast:
-                            if (raycastProvider.Raycast(pointerRays[i], prioritizedLayerMasks, focusIndividualCompoundCollider, out hitInfo))
+                        case SceneQueryType.SphereCast:
+                            if (pointer.SceneQuery(prioritizedLayerMasks, focusIndividualCompoundCollider, out hitInfo))
                             {
                                 hit.Set(hitInfo, pointerRays[i], i, rayStartDistance + hitInfo.distance, focusIndividualCompoundCollider);
-                                return;
                             }
                             break;
                         case SceneQueryType.BoxRaycast:
                             Debug.LogWarning("Box Raycasting Mode not supported for pointers.");
                             break;
-                        case SceneQueryType.SphereCast:
-                            if (raycastProvider.SphereCast(pointerRays[i], pointer.SphereCastRadius, prioritizedLayerMasks, focusIndividualCompoundCollider, out hitInfo))
-                            {
-                                hit.Set(hitInfo, pointerRays[i], i, rayStartDistance + hitInfo.distance, focusIndividualCompoundCollider);
-                                return;
-                            }
-                            break;
                         case SceneQueryType.SphereOverlap:
```

Comment:
Instead of moving _all_ the raycasting calls into the pointers themselves, do you think it makes sense to make a smaller change here and add a custom `SceneQueryType.DefinedInPointer` or something? And then this can do some special sphere pointer logic instead of needing to call into every pointer each time. Thinking through this pattern again, it starts to feel like a bit of a muddying of the division of responsibilities, as some of the "focus" logic is being moved out of the focus provider and into the pointers, instead of the pointers providing the raycast/focus parameters.

Reply:
I always thought that querying the scene should be something that is defined in the pointer itself, rather than inside the focus provider. When the focus provider is making the queries, it feels like the pointer has to be aware of how the focus provider intends to use it's parameters so that they are set correctly. It also would have helped reduces the number of properties that pointers had to share (i.e., the SphereCastRadius ended up being a property in IMixedRealityPointer, where ultimately it feels like it shouldn't have)

There did end up being some redundant code, since some of our older pointers i.e. ggv and generic don't inherit from the BaseControllerPointer class, but I think that it makes the most sense for it to be split up.